src/Simulation/CsharpGeneration/EntryPoint.fs

```diff
@@ -45,36 +45,52 @@ let private constantsClass =
              constant "ResourcesEstimator" "string" (``literal`` AssemblyConstants.ResourcesEstimator)]
         ``}``
 
+/// The name of the C# type used by the parameter in its command-line option, given its Q# type.
+let rec private csharpParameterTypeName context (qsType : ResolvedType) =
+    match qsType.Resolution with
+    | ArrayType itemType -> sprintf "System.Collections.Generic.IEnumerable<%s>"
+                                    (csharpParameterTypeName context itemType)
+    | _ -> SimulationCode.roslynTypeName context qsType
+
 /// A sequence of all of the named parameters in the argument tuple and their respective C# and Q# types.
 let rec private parameters context doc = function
     | QsTupleItem variable ->
-        match variable.VariableName, variable.Type.Resolution with
-        | ValidName name, ArrayType itemType ->
-            // Command-line parsing libraries can't convert to IQArray. Use IEnumerable instead.
-            let typeName = sprintf "System.Collections.Generic.IEnumerable<%s>"
-                                   (SimulationCode.roslynTypeName context itemType)
-            Seq.singleton { Name = name.Value
-                            QsharpType = variable.Type
-                            CsharpTypeName = typeName
-                            Description = ParameterDescription doc name.Value }
-        | ValidName name, _ ->
+        match variable.VariableName with
+        | ValidName name ->
             Seq.singleton { Name = name.Value
                             QsharpType = variable.Type
-                            CsharpTypeName = SimulationCode.roslynTypeName context variable.Type
+                            CsharpTypeName = csharpParameterTypeName context variable.Type
                             Description = ParameterDescription doc name.Value }
-        | InvalidName, _ -> Seq.empty
+        | InvalidName -> Seq.empty
     | QsTuple items -> items |> Seq.map (parameters context doc) |> Seq.concat
 
-/// The custom argument handler for the given Q# type.
-let private argumentHandler =
-    // TODO: Support array types derived from types that need a custom argument handler.
-    function
-    | UnitType -> Some "UnitArgumentHandler"
-    | Result -> Some "ResultArgumentHandler"
-    | BigInt -> Some "BigIntArgumentHandler"
-    | Range -> Some "RangeArgumentHandler"
-    | _ -> None
-    >> Option.map (fun handler -> ``ident`` "Driver" <|.|> ``ident`` handler)
+/// The argument parser for the Q# type.
+let private argumentParser qsType =
+    let rec valueParser = function
+        | ArrayType (itemType : ResolvedType) -> valueParser itemType.Resolution
+        | BigInt -> Some "TryParseBigInteger"
+        | Range -> Some "TryParseQRange"
```

Comment:
Are you missing Pauli here?

Reply:
`Pauli` is an enum, so it is natively supported by System.CommandLine.